reposerver/repository/repository_test.go

```diff
@@ -2208,6 +2208,44 @@ func TestGenerateManifestWithAnnotatedTagsAndMultiSourceApp(t *testing.T) {
 	}
 }
 
+func TestGenerateMultiSourceHelmWithFileParameter(t *testing.T) {
+
+	service := newService(t, "../../util/helm/testdata")
+
+	refSources := map[string]*argoappv1.RefTarget{}
+
+	refSources["$global"] = &argoappv1.RefTarget{
+		TargetRevision: "HEAD",
+	}
+
+	manifestRequest := &apiclient.ManifestRequest{
+		Repo: &argoappv1.Repository{},
+		ApplicationSource: &argoappv1.ApplicationSource{
+			Ref:            "$global",
+			Path:           "./redis",
+			TargetRevision: "HEAD",
+			Helm: &argoappv1.ApplicationSourceHelm{
+				ValueFiles: []string{"$global/redis/values-production.yaml"},
+				FileParameters: []argoappv1.HelmFileParameter{{
+					Name: "password",
+					Path: "$global/external/external-secret.txt",
+				}},
+			},
+		},
+		HasMultipleSources: true,
+		NoCache:            true,
+		RefSources:         refSources,
+	}
+
+	res, err := service.GenerateManifest(context.Background(), manifestRequest)
+	if err != nil {
+		t.Errorf("unexpected %s", err)
+	}
+
+	assert.NoError(t, err)
+	assert.Contains(t, res.Manifests[6], `"replicas":3`, "Values should not be overriden")
```

Comment:
@cardoe Do you think it makes sense to add another assertion here to verify that the `password` parameter is set?

Reply:
Yeah. Struggling a little to know which array position in the Manifests I need to look at.

Reply:
You might want to check out [slices.IndexFunc](https://pkg.go.dev/golang.org/x/exp/slices), if there’s specific values you can match on that could work.

If you want, I can take a look next Monday.

Reply:
Well it turns out I was setting the wrong file parameter. I thought I had set the one that didn't base64 the input. Correcting that helped. I think this is a better test now. But again I'll readily admit I'm not a Go programmer so your input is definitely appreciated.

I have been able to take this branch and build the container and use it for a project where I had a reference to a helm file parameter and it went from erroring to setting the value correctly.

Reply:
I can't open a PR against your branch, not sure why. 

I added one commit at https://github.com/morremeyer/argo-cd/commit/aa090e2112b3e4cf88bffa6c07e5abab8cdc9e74 that uses `slices.IndexFunc` to check against all manifests, not just one. 

Did not run the full test suite against it yet, feel free to add it to this PR.